File: 4/6/main.go
package main

import (
	"fmt"
	"net"
)

func main() {
	listener, err := net.Listen("tcp", ":8080")
	if err != nil {
		panic(err)
	}
	defer listener.Close()

	fmt.Println("Сервер запущен на порту 8080...")

	for {
		conn, err := listener.Accept()
		if err != nil {
			fmt.Println("Ошибка при принятии соединения:", err)
			continue
		}
		go handleConnection(conn)
	}
}

func handleConnection(conn net.Conn) {
	defer conn.Close()

	buffer := make([]byte, 1024)
	_, err := conn.Read(buffer)
	if err != nil {
		fmt.Println("Ошибка при чтении из соединения:", err)
		return
	}
	html := generateHTML()

	response := "HTTP/1.1 200 OK\r\n" +
		"Content-Type: text/html; charset=utf-8\r\n" +
		fmt.Sprintf("Content-Length: %d\r\n", len(html)) +
		"\r\n" + html

	conn.Write([]byte(response))
}

func generateHTML() string {
	studentName := "Кручинин Иван Юрьевич"
	primes := getPrimes(0, 100)

	html := "<html><head><title>Простые числа</title></head><body>"
	html += "<h1>ФИО студента: " + studentName + "</h1>"
	html += "<h2>Простые числа от 0 до 100:</h2><ul>"
	for _, prime := range primes {
		html += fmt.Sprintf("<li>%d</li>", prime)
	}
	html += "</ul></body></html>"
	return html
}

func getPrimes(start, end int) []int {
	var primes []int
	for i := start; i <= end; i++ {
		if isPrime(i) {
			primes = append(primes, i)
		}
	}
	return primes
}

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

File: 4/python/6/main.py
import socket
import threading


def is_prime(n):
    if n < 2:
        return False
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True


def get_primes(start, end):
    return [i for i in range(start, end + 1) if is_prime(i)]


def generate_html():
    student_name = "Кручинин Иван Юрьевич" #todo тут менять фио
    primes = get_primes(0, 100)

    html = "<html><head><title>Простые числа</title></head><body>"
    html += f"<h1>ФИО студента: {student_name}</h1>"
    html += "<h2>Простые числа от 0 до 100:</h2><ul>"
    for prime in primes:
        html += f"<li>{prime}</li>"
    html += "</ul></body></html>"
    return html


def handle_connection(conn, addr):
    print(f"Подключение от {addr}")
    try:
        conn.recv(1024)

        html = generate_html()
        response = (
                "HTTP/1.1 200 OK\r\n"
                "Content-Type: text/html; charset=utf-8\r\n"
                f"Content-Length: {len(html)}\r\n"
                "\r\n" + html
        )
        conn.sendall(response.encode('utf-8'))
    except Exception as e:
        print("Ошибка при обработке соединения:", e)
    finally:
        conn.close()


def main():
    host = ""
    port = 8080
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    server_socket.bind((host, port))
    server_socket.listen(5)
    print(f"Сервер запущен на порту {port}...")

    while True:
        conn, addr = server_socket.accept()
        threading.Thread(target=handle_connection, args=(conn, addr), daemon=True).start()


if __name__ == "__main__":
    main()
